File: common/src/main/kotlin/com/seleneworlds/common/grid/GridApi.kt
package com.seleneworlds.common.grid

open class GridApi(private val grid: Grid) {

    fun getDirectionByName(name: String): Direction {
        return grid.getDirectionByName(name)
            ?: throw IllegalArgumentException("Unknown direction: $name")
    }

    fun defineDirection(name: String, x: Int, y: Int, z: Int, angle: Float): Coordinate {
        return grid.defineDirection(name, Coordinate(x, y, z), angle)
    }
}

File: client-web/src/api/GridApi.ts
import type { Coordinate } from '@/networking/GameProtocol';
export type { ClientGridDefinition, ClientGridDirectionDefinition } from '@/data/ClientRegistrySchemas';

export interface ClientDirection {
  name: string;
  vector: Coordinate;
  angle: number;
}

export interface GridApi {
  defineDirection: (name: string, x: number, y: number, z: number, angle: number) => Coordinate;
  getDirectionByName: (name: string) => ClientDirection;
  getDirection: (angle: number) => ClientDirection | null;
  screenToCoordinate: (x: number, y: number, z?: number) => Coordinate;
}
